pkg/cloudprovider/aws/instancetype.go

```diff
@@ -60,66 +60,99 @@ func (i *InstanceType) Architecture() string {
 	return fmt.Sprint(aws.StringValueSlice(i.ProcessorInfo.SupportedArchitectures)) // Unrecognized, but used for error printing
 }
 
-func (i *InstanceType) CPU() *resource.Quantity {
-	return resources.Quantity(fmt.Sprint(*i.VCpuInfo.DefaultVCpus))
+func (i *InstanceType) Resources() v1.ResourceList {
+	return v1.ResourceList{
+		v1.ResourceCPU:             i.cpu(),
+		v1.ResourceMemory:          i.memory(),
+		v1.ResourcePods:            i.pods(),
+		v1alpha1.ResourceAWSPodENI: i.awsPodENI(),
+		v1alpha1.ResourceNVIDIAGPU: i.nvidiaGPUs(),
+		v1alpha1.ResourceAMDGPU:    i.amdGPUs(),
+		v1alpha1.ResourceAWSNeuron: i.awsNeurons(),
+	}
+}
+
+func (i *InstanceType) Price() float64 {
+	const MbToGb = 1 / 1024.0
+
+	gpuMemoryMiB := 0.0
+	gpuCount := 0.0
+	if i.GpuInfo != nil {
+		for _, gpu := range i.GpuInfo.Gpus {
+			if gpu.MemoryInfo != nil && gpu.MemoryInfo.SizeInMiB != nil {
+				gpuMemoryMiB += float64(*gpu.MemoryInfo.SizeInMiB)
+			}
+			if gpu.Count != nil {
+				gpuCount += float64(*gpu.Count)
+			}
+		}
+	}
+
+	// These are meant to give some relative weighting
+	return float64(*i.VCpuInfo.DefaultVCpus) +
+		float64(*i.MemoryInfo.SizeInMiB)/MbToGb +
+		5*gpuCount + 10*gpuMemoryMiB/MbToGb
```

Comment:
Consider extracting these values to constants

var (
  GPUCostWeight
  CPUCostWeight
  MemoryGBCostWeight
)